app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt

```diff
@@ -6,75 +6,85 @@
+import androidx.navigation.NavController
+import androidx.navigation.Navigation
 import io.reactivex.rxkotlin.addTo
 import mozilla.lockbox.R
 import mozilla.lockbox.action.RouteAction
 import mozilla.lockbox.flux.Presenter
+import mozilla.lockbox.log
 import mozilla.lockbox.store.RouteStore
-import mozilla.lockbox.view.FxALoginFragment
-import mozilla.lockbox.view.ItemListFragment
-import mozilla.lockbox.view.SettingFragment
-import mozilla.lockbox.view.WelcomeFragment
-import mozilla.lockbox.view.LockedFragment
-import mozilla.lockbox.view.ItemDetailFragment
-import mozilla.lockbox.view.FilterFragment
+import mozilla.lockbox.view.ItemDetailFragmentArgs
 
-class RoutePresenter(private val activity: AppCompatActivity, routeStore: RouteStore = RouteStore.shared) : Presenter() {
-    private val welcome: WelcomeFragment by lazy { WelcomeFragment() }
-    private val login: FxALoginFragment by lazy { FxALoginFragment() }
-    private val itemList: ItemListFragment by lazy { ItemListFragment() }
-    private val settingList: SettingFragment by lazy { SettingFragment() }
-    private val lock: LockedFragment by lazy { LockedFragment() }
-    private val itemDetail: ItemDetailFragment by lazy { ItemDetailFragment() }
-    private val filter: FilterFragment by lazy { FilterFragment() }
-
-    init {
-        routeStore.routes
-                .subscribe(this::route)
-                .addTo(compositeDisposable)
-    }
+class RoutePresenter(
+    private val activity: AppCompatActivity,
+    private val routeStore: RouteStore = RouteStore.shared
+) : Presenter() {
+    private lateinit var navController: NavController
 
     override fun onViewReady() {
-        replaceFragment(this.welcome, false)
+        navController = Navigation.findNavController(activity, R.id.fragment_nav_host)
+        routeStore.routes.subscribe(this::route).addTo(compositeDisposable)
     }
 
-    private fun replaceFragment(frag: Fragment, backable: Boolean = true) {
-        val tx = activity.supportFragmentManager.beginTransaction()
-        tx.replace(R.id.root_content, frag)
-        if (backable) {
-            tx.addToBackStack(null)
-        }
-        tx.commit()
-
-        if (!backable) {
-            clearBackStack()
-        }
-    }
-
-    private fun clearBackStack() {
-        val fm = activity.supportFragmentManager
-        if (fm.backStackEntryCount > 0) {
-            val base = fm.getBackStackEntryAt(0)
-            fm.popBackStackImmediate(base.id, FragmentManager.POP_BACK_STACK_INCLUSIVE)
+    private fun replaceFragment(@IdRes destinationId: Int, args: Bundle? = null) {
+        val from = navController.currentDestination?.id ?: return
+        if (from == destinationId && args == null) {
+            // No point in navigating if nothing has changed.
+            return
         }
+        val transition = findTransitionId(from, destinationId) ?: destinationId
+        navController.navigate(transition, args)
```

Comment:
hmm, so just to confirm that I'm reading this correctly... we navigate to the destination even if we don't have a specified transition for it?

Reply:
Correct. If we can't find a transition, then we can still get there, just not with a funky transition. We should have the funky transition. 

Should we `IllegalStateException` if not, to force us to keep the graph in sync?

Reply:
I do like the funky transitions...

Being able to start views that aren't specified as "next" ones for the current view seems like it might be useful for things like deep-linking, though. If we are unlocked and on the settings page, but someone launches the app via some kind of autofill dialog, we still want to show the entries list or a specified entry.

My follow-up question is, what happens to the graph when it is not N'SYNC ("in sync")?

Reply:
There's a whole bunch of support for deep-linking, and putting together a back stack based on a deep-link, it's pretty cool. 

Also: we're going to want to be using multiple graphs, and multiple activities, and be able to navigate around them. It gets suuper complicated quite quickly.

So: in answer to your question, if the `RoutePresenter` is not in sync with the graph, then we just navigate straight to the destination.

I've improved the error message for you though.